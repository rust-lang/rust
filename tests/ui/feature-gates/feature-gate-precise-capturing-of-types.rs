fn foo<T>() -> impl Sized + use<> {}
//~^ ERROR `impl Trait` must mention all type parameters in scope in `use<...>`

fn main() {}
