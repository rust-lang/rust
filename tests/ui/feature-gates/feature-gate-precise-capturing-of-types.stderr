error: `impl Trait` must mention all type parameters in scope in `use<...>`
  --> $DIR/feature-gate-precise-capturing-of-types.rs:1:16
   |
LL | fn foo<T>() -> impl Sized + use<> {}
   |        -       ^^^^^^^^^^^^^^^^^^
   |        |
   |        type parameter is implicitly captured by this `impl Trait`
   |
   = note: currently, all type parameters are required to be mentioned in the precise captures list

error: aborting due to 1 previous error

